Gems/Camera/Code/Source/EditorCameraComponent.cpp

```diff
@@ -155,6 +158,42 @@ namespace Camera
         }
     }
 
+    bool EditorCameraComponent::GetCameraState(AzFramework::CameraState& cameraState)
+    {
+        const CameraComponentConfig& config = m_controller.GetConfiguration();
+        AZ::RPI::ViewportContextPtr viewportContext = m_controller.GetViewportContext();
+        AZ::RPI::ViewPtr view = m_controller.GetView();
+
+        if (viewportContext == nullptr || view == nullptr)
+        {
+            return false;
+        }
+
+        AzFramework::SetCameraTransform(cameraState, view->GetCameraTransform());
```

Comment:
It doesn't look right that a getter function (`GetCameraState`) calls a a setter function to set the camera transform. Why is this needed?

Reply:
The method name is a bit misleading in this context; this is just applying the camera transform to `cameraState` (internally it just decomposes the transform into forward/up/left normals plus a translation)